### PR TITLE
Feature/authentication

### PR DESCRIPTION
**Pull request by ayadjo**

### CODE REVIEW

**Review by vanja4 — APPROVED** [review]

File: Konteh/Konteh.BackOffice.Api/Konteh.BackOffice.Api.csproj [hunk 1]
```diff
@@ -18,7 +18,10 @@
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
+    <PackageReference Include="Microsoft.Identity.Web" Version="3.3.1" />
+    <PackageReference Include="Microsoft.Identity.Web.UI" Version="3.3.1" />
```
Comment (laslorajda) [inline Konteh/Konteh.BackOffice.Api/Konteh.BackOffice.Api.csproj:22]: I don't think you need this package

File: Konteh/Konteh.BackOffice.Api/Konteh.BackOffice.Api.csproj [hunk 2]
```diff
@@ -18,7 +18,10 @@
...
     <PackageReference Include="NSwag.AspNetCore" Version="14.1.0" />
+    <PackageReference Include="Swashbuckle.AspNetCore" Version="7.0.0" />
```
Comment (laslorajda) [inline Konteh/Konteh.BackOffice.Api/Konteh.BackOffice.Api.csproj:24]: I know you don't need this package

File: Konteh/back-office-app/src/app/app.module.ts [hunk 3]
```diff
@@ -1,21 +1,75 @@
...
+import { MatButtonModule } from '@angular/material/button';
+import { MatToolbarModule } from '@angular/material/toolbar';
+import { MatIconModule } from '@angular/material/icon';
+import { HttpClientModule } from '@angular/common/http';
+
+
+export function MSALInstanceFactory(): IPublicClientApplication {
+  return new PublicClientApplication({
+    auth: {
+      clientId: '4e1ff54b-bf34-4f45-83ce-e50fc32967cd',
```
Comment (laslorajda) [inline Konteh/back-office-app/src/app/app.module.ts:21]: parameters like these should always be configurable

File: Konteh/back-office-app/src/app/masl.guard.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  constructor(private authService: MsalService) {
+
+  }
+
+  canActivate(
+    route: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+
+    if (this.authService.instance.getActiveAccount() == null) {
+      console.log('not logged in!')
```
Comment (laslorajda) [inline Konteh/back-office-app/src/app/masl.guard.ts:20]: no console logs in code please

File: Konteh/back-office-app/src/app/navigation-bar/navigation-bar.component.css [hunk 5]
```diff
@@ -0,0 +1,32 @@
+mat-toolbar {
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+  }
+  
+  .spacer {
+    flex: 1; /* Ovaj razmak pomaže u poravnanju dugmadi sa desne strane */
```
Comment (laslorajda) [inline Konteh/back-office-app/src/app/navigation-bar/navigation-bar.component.css:8]: What is this weird language?

File: Konteh/back-office-app/src/app/navigation-bar/navigation-bar.component.css [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  
+  button.mat-button {
+    margin-left: 10px;
+  }
+  
+  button.mat-icon-button {
+    margin-left: 10px;
+  }
+  
+  /* Stil za aktivne linkove */
```
Comment (laslorajda) [inline Konteh/back-office-app/src/app/navigation-bar/navigation-bar.component.css:23]: .

File: Konteh/back-office-app/src/app/navigation-bar/navigation-bar.component.ts [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+      })
+    });
+  }
+
+  isLoggedIn(): boolean {
+    return this.authService.instance.getActiveAccount() != null
+  }
+
+  login() {
+    const loginRequest = {
```
Comment (laslorajda) [inline Konteh/back-office-app/src/app/navigation-bar/navigation-bar.component.ts:32]: I don't think you need this.

**Review by laslorajda — CHANGES_REQUESTED** [review]
some cleanup needed

**Review by laslorajda — APPROVED** [review]

**Pull request merged**